src/management-system-v2/components/phone-input.tsx

```diff
@@ -0,0 +1,83 @@
+'use client';
+
+import { Input, InputProps, Select, SelectProps } from 'antd';
+import {
+  CountryCode,
+  getCountries,
+  getCountryCallingCode,
+  parsePhoneNumberFromString,
+} from 'libphonenumber-js';
+import Image from 'next/image';
+import { useEffect, useState } from 'react';
+
+const getCountryOption = (country: CountryCode) => {
+  const imageUrl = ['SJ', 'AC', 'BQ', 'GF', 'IO', 'GP', 'XK'].includes(country)
+    ? '_unknown'
+    : country;
+
+  return {
+    label: (
+      <span>
+        <Image src={`/flags-32/${imageUrl}.png`} alt={`${country} flag`} width={20} height={20} />{' '}
+        {`${country} +${getCountryCallingCode(country)}`}
+      </span>
+    ),
+    value: country,
+  };
+};
+
+export default function PhoneInput(
+  props: {
+    selectProps?: SelectProps<CountryCode, ReturnType<typeof getCountryOption>>;
+  } & InputProps,
+) {
+  const [country, setCountry] = useState<CountryCode>(() => {
+    if (!props.value || typeof props.value !== 'string') return 'DE';
+
+    return parsePhoneNumberFromString(props.value)?.country || 'DE';
+  });
+  const [value, setValue] = useState(() => {
+    if (!props.value || typeof props.value !== 'string') return '';
+
+    return parsePhoneNumberFromString(props.value)?.nationalNumber || '';
+  });
+
+  useEffect(() => {
+    if (props.onChange) {
+      const code = getCountryCallingCode(country);
+      const phoneNumber = `+${code} ${value}`;
+      const changeEvent = {
+        target: { value: phoneNumber },
+      } as React.ChangeEvent<HTMLInputElement>;
+
+      props.onChange(changeEvent);
+    }
+  }, [country, props.onChange]);
```

Comment:
Can this be moved into the `setCountry` handler below to avoid this effect?

Reply:
You're right, it didn't occur to me